config_proc_macro/src/lib.rs

```diff
@@ -82,3 +83,14 @@ pub fn rustfmt_only_ci_test(_args: TokenStream, input: TokenStream) -> TokenStre
         token_stream
     }
 }
+
+/// Implement the StyleEditionDefault trait for the given item;
+#[proc_macro_attribute]
+pub fn style_edition(args: TokenStream, input: TokenStream) -> TokenStream {
```

Comment:
I think it would make more sense as a derive macro, with the configuration of defaults as a helper attribute.

Reply:
Interesting idea! I use the `#[config_type]` proc macro as a reference, which helps us implement the `ConfigType` trait for all our config options. Are there advantages to defining this as a derive instead of a custom proc macro? would it simplify the implementation?

Reply:
I prefer derives because

1. custom derives make it so that you can't change the original type definition.
2. custom derives make it obvious that you are generating the implementation of a trait, when the naming of the derive macro matches the trait that you are generating impl for.

I'm not sure if it would actually result in any difference in the implementation or simplify anything. It is probably just a stylistic choice here.

Reply:
Thanks for laying out your rational! Given that it's more of a stylistic decision I'd prefer to hold off on changing this from a proc macro to a derive macro until after we've got `style_edition` fully implemented. Then we might consider changing both `#[config_type]` and `#[style_edition]` to custom derives.

Reply:
@fee1-dead do you have any suggestions on improving / simplifying the proposed implementation?